docs/docs/behaviors/macros.md

```diff
@@ -0,0 +1,223 @@
+
+```
+bindings
+    = <&macro_press &kp LSHFT>
+    , <&macro_tap &kp Z &kp M &kp K>
+    , <&macro_release &kp LSHFT>
+    ;
+```
+
+### Processing Continuation on Release
+
+The macro can be paused so that only part of the `bindings` list is processed when the macro is pressed, and the remainder is processed once
+the macro itself is released.
+
+To pause the macro until release, use `&macro_pause_for_release`. For example, this macro will press a modifier and activate a layer when the macro is pressed. Once the macro is released, it will release the modifier and deactivate the layer by releasing the `&mo`:
+
+```
+bindings
+    = <&macro_press &mo 1 &kp LSHFT>
+    , <&macro_pause_for_release>
+    , <&macro_release &mo 1 &kp LSHFT>
+    ;
+```
+
+### Wait Time
+
+The wait time setting controls how long of a delay is introduced between behaviors in the `bindings` list. The initial wait time for a macro, 100ms by default, can
+be set by assigning a value to the `wait-ms` property of the macro, e.g. `wait-ms = <20>;`. If you want to update the wait time at any
+point in the macro bindings list, use `&macro_wait_time`, e.g. `&macro_wait_time 30`. A full example:
+
+```
+wait-ms = <10>;
+bindings
+    = <&kp F &kp A &kp S &kp T>
+    , <&macro_wait_time 500>
+    , <&kp S &kp L &kp O &kp W>
+    ;
+```
+
+### Tap Time
+
+The tap time setting controls how long a tapped behavior is held in the `bindings` list. The initial tap time for a macro, 100ms by default, can
+be set by assigning a value to the `tap-ms` property of the macro, e.g. `tap-ms = <20>;`. If you want to update the tap time at any
+point in a macro bindings list, use `&macro_tap_time`, e.g. `&macro_tap_time 30`. A full example:
+
+```
+bindings
+    = <&macro_tap_time 10>
+    , <&kp S &kp H &kp O &kp R &kp T>
+    , <&macro_tap_time 500>
+    , <&kp L &kp O &kp N &kp G>
+    ;
+```
+
+## Common Patterns
+
+### Layer Activation + More
+
+Macros make it easy to combine a [layer behavior](/docs/behaviors/layers), e.g. `&mo` with another behavior at the same time.
+Common examples are enabling one or more modifiers when the layer is active, or changing the RBG underglow color.
+
+To achieve this, a combination of a 0ms wait time and splitting the press and release between a `&macro_pause_for_release` is used:
+
+#### Layer + Modifier
+
+```
+wait-ms = <0>;
+bindings
+    = <&macro_press &mo 1 &kp LSHFT>
+    , <&macro_pause_for_release>
+    , <&macro_release &mo 1 &kp LSHFT>;
+```
+
+#### Layer + Underglow Color
+
+To trigged a different underglow when the macro is pressed, and when it is released, we use the macro "press" activation mode whenever triggering the `&rgb_ug` behavior:
+
+```
+wait-ms = <0>;
+bindings
+    = <&macro_press   &mo 1              &rgb_ug RGB_COLOR_HSB(128,100,100)>
+    , <&macro_pause_for_release>
+    , <&macro_release &mo 1 &macro_press &rgb_ug RGB_COLOR_HSB(300,100,50)>;
```

Comment:
Is it okay to press &rgb_ug twice but never release it?

Reply:
This is a good point that I think we should include in the documentation.

What is the expected outcome when a key is pressed within a macro but not released via `&macro_release`? Should the macro release any pressed keys upon completion regardless of whether `&macro_release` is used or not?

Reply:
It could potentially be useful to define a macro that never releases a key and have a different macro that does release it, though having keys be pressed and never released in a macro is probably more likely to be accidental.

Reply:
FYI: One of the good use cases for not releasing a key in macro is to keep is opening app switcher on Mac:
```c
ZMK_MACRO(
        app_switcher,
        wait-ms = <0>;
        tap-ms = <0>;
        bindings
            = <&macro_press &kp LGUI>
            , <&macro_tap &kp TAB>
            ;
    )
```
This opens  app switcher and keeps it open. With right programming of other keys, it would release GUI key i.e., on enter key.